ludwig/utils/backward_compatibility.py

```diff
@@ -106,6 +135,54 @@ def _upgrade_feature(feature: Dict[str, Any]):
     _traverse_dicts(feature, _upgrade_use_bias)
 
 
+def _upgrade_encoder_decoder_params(feature: Dict[str, Any], input_feature: bool) -> None:
+    """
+    This function nests un-nested encoder/decoder parameters to conform with the new config structure for 0.6
+    Args:
+        feature (Dict): Feature to nest encoder/decoder params for.
+        input_feature (Bool): Whether this feature is an input feature or not.
+
+    Returns:
+
+    """
+    warn = False
+    if input_feature:
+        module_type = ENCODER
+    else:
+        module_type = DECODER
+
+    module = feature.get(module_type, {})
+    keys = INPUT_FEATURE_KEYS if module_type == ENCODER else OUTPUT_FEATURE_KEYS
+    if isinstance(module, str):
+        module = {TYPE: module}
+        feature[module_type] = module
+        warn = True
+
+    nested_params = []
+    for k, v in feature.items():
+        if k not in keys:
+            module[k] = v
+            nested_params.append(k)
+
+    if module_type in feature:
+        feature[module_type].update(module)
+    else:
+        feature[module_type] = module
+
+    for k in nested_params:
+        del feature[k]
+
+    if len(nested_params) > 0:
+        warn = True
```

Comment:
nit - you can set warn=True right after line 165 (within the if statement) because you know you've added at least one item there already. That way you don't need this explicit check here

Reply:
Right right, I thought of that but felt weird potentially setting it multiple times, I guess it's probably cleaner that way since it's less lines and we don't care if it sets it multiple times.

Reply:
@connor-mccorm Yeah I wasn't sure either, I was thinking the same thing too, but either way is fine really. Nbd at all